### PR TITLE
feat: add recommended field notice

### PR DESCRIPTION
**Pull request by KClough**
**Summary:**

Related #885

**Expected behavior:** 

Adds the recommended field notice.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
~- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java [hunk 1]
```diff
@@ -112,6 +113,14 @@ public String asString(int columnIndex, FieldLevelEnum level) {
...
           new MissingRequiredFieldNotice(
               fileName, row.getRowNumber(), header.getColumnName(columnIndex)));
+    } else if (level == FieldLevelEnum.RECOMMENDED && s == null) {
+      noticeContainer.addValidationNotice(
+          new MissingRecommendedFieldNotice(
+              fileName, row.getRowNumber(), header.getColumnName(columnIndex)));
+    } else if (level == FieldLevelEnum.RECOMMENDED && s == null) {
+      noticeContainer.addValidationNotice(
+          new MissingRecommendedFieldNotice(
+              fileName, row.getRowNumber(), header.getColumnName(columnIndex)));
```
Comment (maximearmstrong) [inline core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java:123]: It seems to be the duplication of the 4 previous lines.
```suggestion
```

**Review by maximearmstrong — CHANGES_REQUESTED** [review]
Thanks @KClough! A few things before approval.

File: core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java [hunk 2]
```diff
@@ -112,6 +113,14 @@ public String asString(int columnIndex, FieldLevelEnum level) {
       noticeContainer.addValidationNotice(
           new MissingRequiredFieldNotice(
               fileName, row.getRowNumber(), header.getColumnName(columnIndex)));
+    } else if (level == FieldLevelEnum.RECOMMENDED && s == null) {
```
[the remark this answers is not in the document]
Comment (KClough) [reply core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java:116]: Good call. I've added some unit tests for both Recommended and Required. Let me know if you'd like me to cover any additional cases.

**Review by maximearmstrong — APPROVED** [review]
LGTM, thank you @KClough!

**Pull request merged**